go/parquet/internal/encoding/plain_encoder_types.gen.go

```diff
@@ -0,0 +1,553 @@
+// Code generated by plain_encoder_types.gen.go.tmpl. DO NOT EDIT.
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encoding
+
+import (
+	"math"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/parquet"
+	"github.com/apache/arrow/go/parquet/internal/utils"
+	"golang.org/x/xerrors"
+)
+
+// PlainInt32Encoder is an encoder for int32 values using Plain Encoding
+// which in general is just storing the values as raw bytes of the appropriate size
+type PlainInt32Encoder struct {
+	encoder
+
+	bitSetReader utils.SetBitRunReader
+}
+
+// Put encodes a slice of values into the underlying buffer
+func (enc *PlainInt32Encoder) Put(in []int32) {
+	enc.append(arrow.Int32Traits.CastToBytes(in))
```

Comment:
Will this work on bigendian?

Reply:
that's actually a good point, i'll update this to fix that but this also means that we should file a JIRA card since the C++ implementation has the same issue currently as it does exactly the same logic and thus would have an issue on big endian (for numeric types it just memcpy's the bytes as is without accounting for big-endian)

Reply:
added `write{{Type}}LE` and `copyFrom{{Type}}LE` functions for numeric types so that the library properly handles BigEndian in addition to LittleEndian now.

Reply:
that is quite possible.  CC @kiszk about big-endian issue in the C++ parquet implementation

Reply:
Yes, I know that C++ implementation does not support big-endian yet. Thank you for supporting big-endian from the beginning. We will follow this implementation to fix C++ implementation.